via/static/scripts/video_player/components/test/Transcript-test.js

```diff
@@ -168,24 +179,89 @@ describe('Transcript', () => {
     }
   });
 
-  it("scrolls transcript when controller's `scrollToCurrentSegment` method is called", () => {
-    const controlsRef = { current: null };
-    const wrapper = mount(
-      <Transcript
-        controlsRef={controlsRef}
-        transcript={transcript}
-        currentTime={5}
-      />
-    );
-    const scrollContainer = wrapper.find('div[data-testid="scroll-container"]');
-    const scrollTo = sinon.spy(scrollContainer.getDOMNode(), 'scrollTo');
+  describe('scrolling methods', () => {
+    let wrappers;
+
+    const longTranscript = {
+      segments: Array(20)
+        .fill(0)
+        .map((_, i) => ({
+          start: i * 10,
+          text: `Text of segment ${i}`,
+        })),
+    };
+
+    function renderTranscript() {
+      const controlsRef = { current: null };
+      const wrapper = mount(
+        <Transcript
+          controlsRef={controlsRef}
+          transcript={longTranscript}
+          currentTime={50}
+        />,
+        { attachTo: container }
+      );
+      const scrollContainer = wrapper.find(
+        'div[data-testid="scroll-container"]'
+      );
+      const scrollTo = sinon.spy(scrollContainer.getDOMNode(), 'scrollTo');
+
+      wrappers.push(wrapper);
 
-    assert.ok(controlsRef.current);
-    controlsRef.current.scrollToCurrentSegment();
+      return { controlsRef, scrollContainer, scrollTo, wrapper };
+    }
+
+    beforeEach(() => {
+      wrappers = [];
+      container.style.height = '400px';
+    });
+
+    afterEach(() => {
+      wrappers.forEach(w => w.unmount());
+    });
+
+    it('scrolls to current segment when `scrollToCurrentSegment` is called', () => {
```

Comment:
The tests for the existing `scrollToCurrentSegment` method have been reworked to enable them to distinguish between scrolling to the current position vs the top or bottom.